packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -4193,6 +4201,20 @@ class SemanticsConfiguration {
     _setFlag(SemanticsFlag.isChecked, value!);
   }
 
+  /// If this node has tristate that can be controlled by the user, whether
+  /// that state is in its mixed state.
+  ///
+  /// Do not call the setter for this field if the owning [RenderObject] doesn't
+  /// have checked/unchecked state that can be controlled by the user.
+  ///
+  /// The getter returns null if the owning [RenderObject] does not have
+  /// mixed checked state.
+  bool? get isCheckStateMixed => _hasFlag(SemanticsFlag.hasCheckedState) ? _hasFlag(SemanticsFlag.isCheckStateMixed) : null;
+  set isCheckStateMixed(bool? value) {
+    _setFlag(SemanticsFlag.hasCheckedState, true);
```

Comment:
Should this modify `SemanticsFlag.isChecked`? If no, why not?

Reply:
Also, should `isChecked` modify `SemanticsFlag.isCheckStateMixed`?

Reply:
In all cases I can imagine, they should be mutually exclusive when `true`, so the question is just whether we want to enforce that internally in the `SemanticsConfiguration`

Reply:
Gotcha. Consider adding assertions then: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#use-asserts-liberally-to-detect-contract-violations-and-verify-invariants

Reply:
The thing is about asserting that exclusivity is that it would create an issue for the tests that ensure all flags are supported, as they work by setting all accessibility flags and making sure they are each reflected in the resulting semantics. If we do make this assertion, those tests would need to change in a manner that I'm not yet sure on how.

Reply:
Agreed that generally we should be making liberal use of assertions to assert any invariants, especially since they're documented invariants.

Can you point to tests that would be broken by this? Could be it's the tests that need updating now that we're modelling new state.